File: Python/problem005.py
#What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20 ?7

multiple  = int(1)
for i in range(2,21):
    isPrime= 1
    for j in range(2,int(i**0.5)+1):
        if i % j == 0:
            isPrime = 0
    if isPrime:
        k = 1
        while i**k<20:
            k += 1
        multiple *= i**(k-1)

print(multiple)
            

File: C/problem005.c
/* What is the smallest positive number that is evenly divisible by all of the numbers from 1 to 20 ?7 */
#include <stdio.h>
int main()
{
    int i,j;
    int product = 1;
    for (i = 2; i <= 20;i++)
    {
        int isPrime = 1;
        for (j = 2; j < i;j++)
        {
            if (i%j==0)
            {
                isPrime = 0;
                break;
            }
        }
        if (isPrime)
        {
            int maxpowerofprime = 1;
            while (maxpowerofprime*i < 20)
                maxpowerofprime *= i;
            product *= maxpowerofprime;
        }   
    }
    printf("%d", product);
    return 0;
}